### PR TITLE
fix: update for minimum of Python 3.7

### PR DESCRIPTION
**Pull request by jdblischak**
### Description

**tl;dr** As of fd5daaeff070f9987dba411a0f5262c533a2f666, snakemake requires Python 3.7. This PR updates any reference to previous versions

I came to the GitHub repo to determine the minimum required Python version to run Snakemake, and I quickly got confused. The README displays Python 3.5:

Screen Shot 2022-03-22 at 2 48 08 PM

But `setup.py` clearly requires Python 3.7+

https://github.com/snakemake/snakemake/blob/5f7170e095a961ce11ac87467a20879dc4f220f4/setup.py#L14-L16

I was going to simply update the metadata

https://github.com/snakemake/snakemake/blob/5f7170e095a961ce11ac87467a20879dc4f220f4/setup.py#L109

But I saw in commit 0357911cc919a1289004bad9144e75d2880665c7 that the minimum required version is specified in multiple other places. This PR is my attempt to update the code and tests to remove any reference to Python versions < 3.7.

A few other notes:

* In searching old Issues, I found #910. Now that Python 3.5 is no longer supported, I think it can be closed
* I didn't update any of the tutorials in the documentation since technically a user could install older versions of Snakemake and Python

### QC

I don't think this PR requires additional tests or documentation. Please let me know if there is more you'd like me to do.

* [ ] The PR contains a test case for the changes or the changes are already covered by an existing test case.
* [ ] The documentation (`docs/`) is updated to reflect the changes or this is not necessary (e.g. if the change does neither modify the language nor the behavior or functionalities of Snakemake).

### CODE REVIEW

**Review by johanneskoester — APPROVED** [review]

**Comment by johanneskoester** [comment]
Thanks a lot!
